ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -845,6 +852,91 @@ namespace OrdinalAxis {
             return groupPositions;
         };
 
+        /**
+         * Get index of point inside the extended ordinal positions array.
+         *
+         * @private
+         * @function Highcharts.Axis#getIndexOfPoint
+         *
+         * @param {number} val
+         *        The pixel value of a point.
+         *
+         * @param {Array<number>} [ordinalPositions]
+         *        An array of points visible in the current plot area.
+         *
+         * @param {Array<number>} [extendedOrdinalPositions]
+         *        An array of all points available on the axis
+         *        for the given data set.
+         *
+         * @return {number}
+         */
+        axisProto.getIndexOfPoint = function (
+            val: number,
+            ordinalPositions: Array<number>,
+            extendedOrdinalPositions: Array<number>
+        ): number {
+            const axis = this,
+                // Distance in pixels between two points
+                // on the ordinal axis in the current zoom.
+                ordinalPointPixelInterval = axis.len / ordinalPositions.length - 1,
```

Comment:
if axis.min and axis.max don't overlap values from extendedOrdinalPositions, this is inaccurate. Apart from the error below. I think that this whole thing should be calculated relative to some point from ordinalPositions.